src/imagery/i.eodag/testsuite/test_eodag.py

```diff
@@ -0,0 +1,313 @@
+S2A_MSIL1C_20220202T160501_N0400_R097_T17SPV_20220202T195450 2022-02-02T16:05:01 18% S2MSI1C
+S2A_MSIL1C_20220212T160401_N0400_R097_T17SPV_20220212T195334 2022-02-12T16:04:01 29% S2MSI1C
+S2B_MSIL1C_20220408T155819_N0400_R097_T17SPV_20220408T195945 2022-04-08T15:58:19 29% S2MSI1C"""
+        i_eodag = Module(
+            "i.eodag",
+            flags="l",
+            provider="peps",
+            producttype="S2_MSI_L1C",
+            start="2022-01-01",
+            end="2022-06-01",
+            clouds=50,
+            map="durham",
+            minimum_overlap=70,
+            quiet=True,
+            stdout_=PIPE,
+        )
+        self.assertEqual(i_eodag.outputs["stdout"].value.strip(), output)
+
+    def test_minimum_overlap_b(self):
+        """Test"""
+        # i.eodag -lb provider=peps producttype=S2_MSI_L1C start=2022-05-01 end=2022-06-01 clouds=50 map=durham minimum_overlap=70
+        output = r"""S2B_MSIL1C_20220528T155819_N0400_R097_T17SPV_20220528T194637 2022-05-28T15:58:19  1% S2MSI1C"""
+        i_eodag = Module(
+            "i.eodag",
+            flags="lb",
+            provider="peps",
+            producttype="S2_MSI_L1C",
+            start="2022-05-01",
+            end="2022-06-01",
+            clouds=50,
+            map="durham",
+            minimum_overlap=70,
+            quiet=True,
+            stdout_=PIPE,
+        )
+        self.assertEqual(i_eodag.outputs["stdout"].value.strip(), output)
+
+    def test_area_relation(self):
+        """Test"""
+        output = r"""S2B_MSIL1C_20220528T155819_N0400_R097_T17SPV_20220528T194637 2022-05-28T15:58:19  1% S2MSI1C
+S2A_MSIL1C_20220602T155831_N0400_R097_T17SPV_20220602T211840 2022-06-02T15:58:31  3% S2MSI1C
+S2B_MSIL1C_20220617T155829_N0400_R097_T17SPV_20220618T113811 2022-06-17T15:58:29  8% S2MSI1C"""
+        i_eodag = Module(
+            "i.eodag",
+            flags="lb",
+            provider="peps",
+            producttype="S2_MSI_L1C",
+            start="2022-05-01",
+            end="2022-07-01",
+            clouds=50,
+            area_relation="Contains",
+            limit=3,
+            map="durham",
+            quiet=True,
+            stdout_=PIPE,
+        )
+        self.assertEqual(i_eodag.outputs["stdout"].value.strip(), output)
+
+    def test_save_result_geojson(self):
+        """Test"""
+        # i.eodag -l map=durham provider=peps producttype=S2_MSI_L1C start=2022-05-01 end=2022-06-01 save=search_s2.geojson
+        i_eodag = Module(
+            "i.eodag",
+            flags="l",
+            provider="peps",
+            producttype="S2_MSI_L1C",
+            start="2022-05-01",
+            end="2022-06-01",
+            map="durham",
+            save="results/search_s2.geojson",
+            quiet=True,
+            overwrite=True,
+        )
+        self.assertFileExists("results/search_s2.geojson")
+        self.assertFilesEqualMd5("results/search_s2.geojson", "data/search_s2.geojson")
```

Comment:
This is a too strict test I would argue, as even changes in provider internal references (that are part of the GeoJSON) would cause the test to fail...